### ISSUE TITLE
seeting the flag using move cause entry to be duplicated

### ISSUE THREAD

**Issue by derek-zhou** [djcb/mu#2424]
There is something weird wrt. move:
```
;; mu> (view :docid 70415)
[177](:view (:path "/home/derek/Maildir/cur/1674684838.H152974P1147.mail.3qin.us:2," :size 8968 :changed (25553 43637 0) :date (25553 43428 0) :flags (unread) :from ((:email "alert@uptimerobot.com" :name "UptimeRobot")) :message-id "1674684836167.392514@uptimerobot.com" :priority normal :subject "Monitor is DOWN: mail" :to ((:email "derek@3qin.us")) :maildir "/" :docid 70415))
;; mu> (view :docid 70416)

** (process:1311): CRITICAL **: 17:21:31.627: xapian_try: xapian error 'Document 70416 not found'
[3a](:error 32828 :message "failed to find message for view")
```

Now everything is right; last mail is 70415, unread. If I mark it as `seen`, then:
```
;; mu> (move :docid 70415 :flags "+S")
[187](:update (:size 8968 :date (25553 43428 0) :from ((:email "alert@uptimerobot.com" :name "UptimeRobot")) :message-id "1674684836167.392514@uptimerobot.com" :priority normal :subject "Monitor is DOWN: mail" :to ((:email "derek@3qin.us")) :path "/home/derek/Maildir//cur/1674684838.H152974P1147.mail.3qin.us:2,S" :changed (25553 43900 0) :flags (seen) :maildir "/" :docid 70415) :maybe-view t)
;; mu> (view :docid 70416)

** (process:1311): CRITICAL **: 17:21:59.707: xapian_try: xapian error 'Document 70416 not found'
[3a](:error 32828 :message "failed to find message for view")
```

Still 70416 does not exist; however, if you squint really hard, you will notice that the path of 70415 actually changed, not just the file name. It is at:  `home/derek/Maildir//cur/...` now, please note the double slash between `Maildir` and `cur`. The strange path itself is not a big problem; because double slash is legal. However, if you do index:
```
;; mu> (index)
;; mu> [46](:info index :status running :checked 70537 :updated 1 :cleaned-up 0)
[47](:info index :status complete :checked 70537 :updated 1 :cleaned-up 0)

;; mu> (view :docid 70416)
[176](:view (:path "/home/derek/Maildir/cur/1674684838.H152974P1147.mail.3qin.us:2,S" :size 8968 :changed (25553 43900 0) :date (25553 43428 0) :flags (seen) :from ((:email "alert@uptimerobot.com" :name "UptimeRobot")) :message-id "1674684836167.392514@uptimerobot.com" :priority normal :subject "Monitor is DOWN: mail" :to ((:email "derek@3qin.us")) :maildir "/" :docid 70416))
;; mu> (view :docid 70415)
[177](:view (:size 8968 :date (25553 43428 0) :from ((:email "alert@uptimerobot.com" :name "UptimeRobot")) :message-id "1674684836167.392514@uptimerobot.com" :priority normal :subject "Monitor is DOWN: mail" :to ((:email "derek@3qin.us")) :path "/home/derek/Maildir//cur/1674684838.H152974P1147.mail.3qin.us:2,S" :changed (25553 43900 0) :flags (seen) :maildir "/" :docid 70415))
```
Now there are 2 duplicated entries, 70415 with the double slash, and 70416 with the single slash, and they all point to the same file.

My expectation is that commands like `add` `move` will update the database accordingly, and not to add extraneous slash before `cur`, so `index` will be an no-op and will not resurface the entry without the extra slash. I am using almost the git HEAD.

**Comment by derek-zhou** [comment]
Hi DIrk, have you got a chance to look into this? I'd like to help; however debugging C++ is beyond my capability. I can read and reason about the code and that's about al can do in C++.

**Comment by djcb** [comment]
Sure, I'll take a look.

**Comment by djcb** [comment]
Can't exactly reproduce, but  I've pushed a few changes to master... does that help?

**Comment by derek-zhou** [comment]
Thank. Latest git HEAD is good. Reading the recent git log, it seems like the fix probably come in from 25635c5cfec87df9338a696b17f1f86fa762781e but I am not sure.
